cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -371,6 +396,45 @@ private boolean isAllReplicasReady(Object resource) {
     }
   }
 
+  /**
+   * Checks if job is ready.
+   */
+  private boolean isJobReady(Object resource) {
+    try {
+      Method getStatus = resource.getClass().getDeclaredMethod("getStatus");
```

Comment:
why do we need to do it this way instead of calling the method normally?

Reply:
I followed existing pattern for checking replica readiness. Using k8s api now.